lib/elixir/lib/gen_server.ex

```diff
@@ -566,6 +566,10 @@ defmodule GenServer do
   """
   @spec call(server, term, timeout) :: term
   def call(server, request, timeout \\ 5000) do
+    if server == self do
+      raise ArgumentError, message: 
+      "invalid server argument in call(server, request, timeout). Looks like you trying to call a parent process"
```

Comment:
Let's go with the solution I posted then. :)  Then we can format the `:calling_self` atom here: https://github.com/elixir-lang/elixir/blob/master/lib/elixir/lib/exception.ex#L233 Could you please do the changes @Troush ? Thank you!